arrows/vtk/applets/color_mesh.cxx

```diff
@@ -398,13 +427,56 @@ class color_mesh::priv : public kwiver::arrows::vtk::mesh_coloration
       writer->Write();
       return true;
     }
+    else if (ext == ".las")
+    {
+      save_mesh_las(mesh, output_path);
+      return true;
+    }
     else
     {
       LOG_ERROR(main_logger, "Invalid file format: " << ext);
       return false;
     }
   }
 
+  void save_mesh_las(vtkSmartPointer<vtkPolyData> mesh,
+                     char const * output_path)
+  {
+    auto lgcs = vital::local_geo_cs();
+    read_local_geo_cs_from_file(lgcs, input_geo_origin_file_);
+
+    vtkSmartPointer<vtkPoints> inPts = mesh->GetPoints();
+    vtkIdType numPts = inPts->GetNumberOfPoints();
+    std::string colorArrayName = mesh->GetPointData()->GetScalars()->GetName();
+    vtkSmartPointer<vtkDataArray> da = mesh->GetPointData()->GetArray(colorArrayName.c_str());
+    vtkUnsignedCharArray* rgbArray = nullptr;
+
+    std::vector<vital::vector_3d> points(numPts);
+    std::vector<vital::rgb_color> colors;
+
+    if (da->GetNumberOfComponents() == 3)
+    {
+      rgbArray = vtkArrayDownCast<vtkUnsignedCharArray>(da);
+      colors.resize(numPts);
+    }
+
+    for (vtkIdType i = 0; i < numPts; ++i)
+    {
+      inPts->GetPoint(i, points[i].data());
+
+      if (rgbArray)
+      {
+        vtkIdType idx = 3 * i;
+        colors[i] = vital::rgb_color(rgbArray->GetValue(idx),
+                                     rgbArray->GetValue(idx + 1),
+                                     rgbArray->GetValue(idx + 2));
+      }
+    }
+
+    auto pointcloud_writer = new kwiver::arrows::pdal::pointcloud_io();
```

Comment:
again, memory leak, but allocation not needed if you refactor as described above.